File: practical4/MatchingProfilesA.c
/* match two DNA profiles represented using two one-dimensional arrays

@author J Hong
*/

#include<stdio.h>
#include <stdbool.h>

int main(){
	   
	int size = 10;   	 	
	float suspect[size]; // declaring suspect array
    float criminal[size]; // declaring criminal array

	// read 10 input values into suspect array from keyboard
	printf("Enter the 10 gene chromosomes of the suspect separated by spaces: \n");
	for (int i = 0; i < size; i++)
	  scanf(" %f", &suspect[i]);

    // read 10 input values into criminal array from keyboard
	printf("Enter the 10 gene chromosomes of the criminal separated by spaces: \n");
	for (int i = 0; i < size; i++)
	  scanf(" %f", &criminal[i]);

    // match two profiles
    bool match = true;
    for (int i = 0; i < size; i++) 
	  if (suspect[i] != criminal[i])
      match = false;

    // display matching result
	if (match) 
      printf("The two profiles match! \n");
    else
      printf("The two profiles don't match! \n");
    
	return 0; 
}

File: Practical 6/MatchingProfilesA.py
# match two DNA profiles represented using two one-dimensional arrays
# Practical 6, Part 4
# author R Abdur

size = 10
suspect = []  # declaring suspect list
criminal = []  # declaring criminal list

# read 10 input values into the suspect list from the keyboard

print("\n Enter the 10 gene chromosomes of the suspect one at a time and press Enter\n")
for i in range(size):
    element = float(input("Enter a real number: "))
    suspect.append(element)


# read 10 input values into the criminal list from the keyboard

print("\n Enter the 10 gene chromosomes of the criminal one at a time and press Enter\n")
for i in range(size):
    element = float(input("Enter a real number: "))
    criminal.append(element)

# match two profiles
match = True

for i in range(size):
    if (suspect[i] != criminal[i]):
        match = False

# display matching result

if (match):
    print("\n The two profiles match! \n")
else:
    print("\n The two profiles don't match! \n")
